react-catalog/my-app/src/Details.tsx

```diff
@@ -0,0 +1,54 @@
+import { Link, useNavigate, useParams } from 'react-router-dom';
+import { readProduct } from './read';
+import { Product } from './data';
+import { useEffect, useState } from 'react';
+
+export function Details() {
+  const [product, setProduct] = useState<Product>();
+  const { productId } = useParams();
+  const navigate = useNavigate();
+  useEffect(() => {
+    async function productCall() {
+      try {
+        const currentProduct = await readProduct(Number(productId));
+        setProduct(currentProduct);
+      } catch (error) {
+        console.error(error);
+      }
+    }
+    productCall();
+  }, [productId]);
+  return (
+    <>
+      <div className="flex flex-col border-2 rounded-b-lg p-4">
+        <Link to="/" className="self-start">
+          {' '}
```

Comment:
I'm not sure these blank spaces are needed. I'm also not certain the text needs to be put inside `{}`. You can display `<` in HTML using `&lt;`